client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,366 @@
+				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
+			if is_folder and perm[3] == 'r' and perm[5] == 'x':
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+
+	def signal_menu_toggled_hidden_files(self, _):  # pylint: disable=method-hidden
+		self.local_hidden = not self.local_hidden
+		self.refresh()
+
+	def refresh(self):
+		model = self._tvmodel
+		exp_lines = []
+		model.foreach(lambda model, path, iter: exp_lines.append(path) if self.treeview_local.row_expanded(path) else 0)
+		self.treeview_local.collapse_all()
+		for path in exp_lines:
+			self.treeview_local.expand_row(path, False)
+
+	def signal_menu_activate_delete_prompt(self, _):
+		selection = self.treeview_local.get_selection()
+		model, treeiter = selection.get_selected()
+		dialog = Gtk.Dialog('Warning')
+		label = "Are you sure\n you want to delete this {0}?\n".format('directory' if model[treeiter][5] == -1 else 'file')
+		label = Gtk.Label(label)
+		label.set_justify(Gtk.Justification.CENTER)
+		dialog.vbox.pack_start(label, True, True, 0)
+		label.show()
+		dialog.add_button(Gtk.STOCK_CANCEL, Gtk.ResponseType.REJECT)
+		dialog.add_button(Gtk.STOCK_OK, Gtk.ResponseType.ACCEPT)
+		response_id = dialog.run()
+		dialog.destroy()
+		if response_id == Gtk.ResponseType.ACCEPT:
+			self.delete(model, treeiter)
+
+class LocalDirectory(DirectoryBase):
+	treeview_name = 'treeview_local'
+	def __init__(self, builder, logger, upload_button):
+		_treeview = builder.get_object('SFTPClientGUI.' + self.treeview_name)
+		super(LocalDirectory, self).__init__(_treeview, logger)
+		self.upload_button = upload_button
+		init_dir = os.getcwd()
+		self.load_dirs(init_dir)
+		self.treeview_local.set_model(self._tvmodel)
+
+	def _check_perm(self, fullname):
+		r_permissions = os.access(fullname, os.R_OK)
+		w_permissions = os.access(fullname, os.W_OK)
+		x_permissions = os.access(fullname, os.X_OK)
+		perm = '   '
+		perm = perm + 'r' if r_permissions else perm + '-'
+		perm = perm + 'w' if w_permissions else perm + '-'
+		perm = perm + 'x' if x_permissions else perm + '-'
+		return perm
+
+	def _yield_dir_list(self, path):
+		for name in os.listdir(path):
+			yield(name)
+
+	def _get_raw_time(self, fullname):
+		return os.path.getmtime(fullname)
+
+	def _get_is_folder(self, fullname):
+		return stat.S_ISDIR(os.stat(fullname).st_mode)
+
+	def _get_file_size(self, fullname):
+		return os.path.getsize(fullname)
+
+	def delete(self, model, treeiter):
+		try:
+			if model[treeiter][5] == -1:
+				shutil.rmtree(model[treeiter][2])
+			else:
+				os.remove(model[treeiter][2])
+			self.logger.log_generic('Successfully deleted ' + model[treeiter][2])
+			self.refresh()
```

Comment:
This and the line above it should not be within the exception handler, they would fit better in the `else:` clause of a the try except block.